test/integration/resources/init_timers.c

```diff
@@ -0,0 +1,153 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <rtems.h>
+#include <rtems/bspIo.h>
+#include <bsp/irq.h>
+
+#define IRQMP_BASE 0x80000200U
+
+#define IRQ_LEVEL(n) (((IRQMP_BASE) >> n) | 1U)
+
+#define MAX_TLS_SIZE RTEMS_ALIGN_UP( 64, RTEMS_TASK_STORAGE_ALIGNMENT )
+
+#define TASK_ATTRIBUTES RTEMS_DEFAULT_ATTRIBUTES
+
+#define TASK_STORAGE_SIZE \
+  RTEMS_TASK_STORAGE_SIZE( \
+    MAX_TLS_SIZE + RTEMS_MINIMUM_STACK_SIZE, \
+    RTEMS_DEFAULT_ATTRIBUTES \
+  )
+
+#define CONFIGURE_APPLICATION_NEEDS_CLOCK_DRIVER
+
+#define CONFIGURE_MAXIMUM_TASKS 3
+
+#define CONFIGURE_MINIMUM_TASKS_WITH_USER_PROVIDED_STORAGE CONFIGURE_MAXIMUM_TASKS
+
+#define CONFIGURE_MICROSECONDS_PER_TICK 1000
+
+#define CONFIGURE_MAXIMUM_FILE_DESCRIPTORS 0
+
+#define CONFIGURE_DISABLE_NEWLIB_REENTRANCY
+
+#define CONFIGURE_APPLICATION_DISABLE_FILESYSTEM
+
+#define CONFIGURE_MAXIMUM_THREAD_LOCAL_STORAGE_SIZE MAX_TLS_SIZE
+
+#define CONFIGURE_INTERRUPT_STACK_SIZE RTEMS_ALIGN_UP((4 * CPU_STACK_MINIMUM_SIZE) + CONTEXT_FP_SIZE, CPU_STACK_ALIGNMENT)
+
+#define CONFIGURE_RTEMS_INIT_TASKS_TABLE
+
+#define CONFIGURE_INIT_TASK_ATTRIBUTES RTEMS_DEFAULT_ATTRIBUTES
+
+#define CONFIGURE_INIT_TASK_INITIAL_MODES (RTEMS_DEFAULT_MODES | RTEMS_INTERRUPT_LEVEL(0))
+
+#define CONFIGURE_INIT_TASK_CONSTRUCT_STORAGE_SIZE TASK_STORAGE_SIZE
+
+#define CONFIGURE_INIT
+
+#define TIMER1_CORE_ADDRESS 0x80000300U
+#define TIMER2_CORE_ADDRESS 0x80100600U
+#define TIMER_BASE_ADDRESS 0x10U
+#define TIMER_SIZE 0x10U
+#define TIMEOUT 100000
+#define TIMER_COUNTER_UNDERFLOW_VALUE 0xFFFFFFFFU
```

Comment:
Are you sure this is the correct underflow value? Where is it documented?

Reply:
GR712RC User’s Manual, chapter 12, section 2 Operation, paragraph 2 (page 92). Value changed from 0xFFFFFFFFU to -1U for a more precise reference to the documentation.